### PR TITLE
 Use smaller and compressed varients of buttons and form components

### PR DESCRIPTION
**Pull request by AMoo-Miki**
### Description
Replace instances of `EuiButton` that don't have an explicit sizing attribute to `EuiSmallButton*`.
Replace instances of `Eui<form elements>` that don't have density attributes to `EuiCompressed<form elements>`.

### Check List
- [X] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by SuZhou-Joe** [comment]
https://github.com/opensearch-project/index-management-dashboards-plugin/blob/main/cypress/integration/plugins/index-management-dashboards-plugin/aliases.js#L88

**Comment by AMoo-Miki** [comment]
I have resolved conflicts. Please prioritize reviewing and merging this PR.

File: public/pages/CreateSnapshotPolicy/containers/CreateSnapshotPolicy/CreateSnapshotPolicy.tsx [hunk 1]
```diff
@@ -6,25 +6,25 @@
...
-  EuiButtonEmpty,
-  EuiButton,
+  EuiSmallButtonEmpty,
+  EuiSmallButton,
   EuiComboBoxOptionOption,
-  EuiFieldNumber,
+  EuiCompressedFieldNumber,
   EuiAccordion,
-  EuiRadioGroup,
+  EuiCompressedRadioGroup,
```
Comment (skumawat2025) [inline public/pages/CreateSnapshotPolicy/containers/CreateSnapshotPolicy/CreateSnapshotPolicy.tsx:22]: ```EuiCompressedRadioGroup``` has been imported twice. Can we please remove this.

File: public/pages/SnapshotPolicies/containers/SnapshotPolicies/SnapshotPolicies.tsx [hunk 2]
```diff
@@ -12,7 +12,7 @@ import {
   Criteria,
   Direction,
   EuiBasicTable,
-  EuiButton,
+  EuiSmallButton,
```
Comment (skumawat2025) [inline public/pages/SnapshotPolicies/containers/SnapshotPolicies/SnapshotPolicies.tsx:15]: ```EuiSmallButton1``` has been imported twice. Can we please remove this.

**Comment by SuZhou-Joe** [comment]
Changed the Cypress test case a little bit to make it pass in this PR.

**Review by SuZhou-Joe — APPROVED** [review]

**Pull request merged**